src/components/ShoppingCart/ShoppingCartContext.tsx

```diff
@@ -0,0 +1,100 @@
+import { ReactNode, createContext, useContext, useState } from 'react';
+import ShoppingCart from './ShoppingCart';
+
+type CartItem = {
+	id: number;
+	quantity: number;
+};
+type ShoppingCartContext = {
+	getItemQuantity: (id: number) => number;
+	increaseCartQuantity: (id: number) => void;
+	decreaseCartQuantity: (id: number) => void;
+	removeFromCart: (id: number) => void;
+	openCart: () => void;
+	closeCart: () => void;
+	cartQuantity: number
+	musicCds: CartItem[]
+};
+type ShoppingCartProviderProps = {
+	children: ReactNode;
+};
+
+const ShoppingCartContext = createContext({} as ShoppingCartContext);
+export function useShoppingCart() {
+	return useContext(ShoppingCartContext);
+}
+
+export function ShoppingCartProvider({ children }: ShoppingCartProviderProps) {
+	const [musicCds, setMusicCds] = useState<CartItem[]>([]);
+	const [isOpen, setIsOpen] = useState(false);
+
+	const openCart = () => setIsOpen(true);
+
+	const closeCart = () => setIsOpen(false);
+
+	const cartQuantity = musicCds.reduce(
+		(quantity, item) => item.quantity + quantity,
+		0
+	);
+
+	const increaseCartQuantity = (id: number) => {
+		setMusicCds((currCds) => {
+			if (currCds.find((item) => item.id === id) == null) {
```

Comment:
It's good practice to take such logic to a separate function whose name will indicate what it does